examples/todomvc/src/store.rs

```diff
@@ -0,0 +1,321 @@
+use crate::dbg;
+use js_sys::JSON;
+use wasm_bindgen::prelude::*;
+
+/// Stores items into localstorage
+pub struct Store {
+    local_storage: web_sys::Storage,
+    data: Option<ItemList>,
+    name: String,
+}
+
+impl Store {
+    /// Creates a new store with `name` as the localstorage value name
+    pub fn new(name: &str) -> Option<Store> {
+        if let Some(window) = web_sys::window() {
+            if let Ok(Some(local_storage)) = window.local_storage() {
+                Some(Store {
+                    local_storage,
+                    data: None,
+                    name: String::from(name),
+                })
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    }
+
+    /// Read the local ItemList from localStorage.
+    /// Returns an &Option<ItemList> of the stored database
+    /// Caches the store into `self.data` to reduce calls to JS
+    ///
+    /// Uses mut here as the return is something we might want to manipulate
+    ///
+    fn get_local_storage(&mut self) -> &Option<ItemList> {
+        let mut item_list = ItemList::new();
+        if let Some(ref _data) = self.data {
+            return &self.data;
+        }
+        if let Ok(Some(value)) = self.local_storage.get_item(&self.name) {
```

Comment:
Heh This is definitely a case where we probably want to use `?` :)

The outer function here can return `Option<&ItemList>` and that should enable `?` to be used everywhere here.

Reply:
(either that or I think some `and_then` and/or early returns/continues should be used to avoid the indentation)